### PR TITLE
UCS/STATS: Futexes are only available on Linux.

### PR DESCRIPTION
**Pull request by kostikbel**
Use usual cond var to wait for condition on platforms without futexes.

Signed-off-by: Konstantin Belousov <konstantinb@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: src/ucs/stats/stats.c [hunk 1]
```diff
@@ -496,13 +506,26 @@ static void* ucs_stats_thread_func(void *arg)
...
+#ifdef HAVE_LINUX_FUTEX_H
     flags = ucs_stats_context.flags;
     while (flags & UCS_STATS_FLAG_ON_TIMER) {
         /* Wait for timeout/wakeup */
         ucs_sys_futex(&ucs_stats_context.flags, FUTEX_WAIT, flags, ptime, NULL, 0);
         ucs_stats_dump();
         flags = ucs_stats_context.flags;
     }
+#else
+    pthread_mutex_lock(&ucs_stats_context.lock);
```
Comment (yosefe) [inline src/ucs/stats/stats.c]: pls add comment - TODO  use condvar on Linux as well

File: src/ucs/stats/stats.c [hunk 2]
```diff
@@ -496,13 +506,26 @@ static void* ucs_stats_thread_func(void *arg)
...
+    flags = ucs_stats_context.flags;
+    while (flags & UCS_STATS_FLAG_ON_TIMER) {
+        /* Wait for timeout/wakeup */
+        pthread_cond_timedwait(&ucs_stats_context.cv, &ucs_stats_context.lock,
+	  ptime);
```
Comment (yosefe) [inline src/ucs/stats/stats.c]: indentation

**Comment by yosefe** [comment]
rnat
bot:bgate:retest

File: src/ucs/stats/stats.c [hunk 3]
```diff
@@ -77,6 +82,9 @@ static ucs_stats_context_t ucs_stats_context = {
     .root_node        = {},
     .root_filter_node = {},
     .lock             = PTHREAD_MUTEX_INITIALIZER,
+#ifndef HAVE_LINUX_FUTEX_H
+    .cv		      = PTHREAD_COND_INITIALIZER,
```
Comment (yosefe) [inline src/ucs/stats/stats.c]: indentation

File: src/ucs/stats/stats.c [hunk 4]
```diff
@@ -69,6 +71,9 @@ typedef struct {
     khash_t(ucs_stats_cls) cls;
 
     pthread_mutex_t      lock;
+#ifndef HAVE_LINUX_FUTEX_H
+    pthread_cond_t	 cv;
```
Comment (yosefe) [inline src/ucs/stats/stats.c]: still wrong indentation:
- use spaces instead of tabs

**Pull request merged**